src/components/Modals/EditExpertProfileModal.tsx

```diff
@@ -0,0 +1,166 @@
+  const [email, setEmail] = useState(null);
+  const [expertise, setExpertise] = useState(null);
+  const [description, setDescription] = useState(null);
+  const [expertiseTags, setExpertiseTags] = useState(null);
+  const [visibility, setVisibility] = useState(null);
+
+  const [fieldsMissing, setFieldsMissing] = useState(false);
+
+  const apiContext = useContext(Context.Api);
+  const { user } = useContext(Context.User);
+  const modalContext = useContext(ModalContext);
+
+  const [Avatar, setAvatar] = useState(props.avatar);
+
+  const avatarUploader = useRef(null);
+
+  function closeModal() {
+    modalContext.setOpenedModal(null);
+    // can be optional
+    [
+      setEmail,
+      setExpertise,
+      setDescription,
+      setExpertiseTags,
+      setVisibility,
+    ].map((method) => typeof method === 'function' && method(null));
+  }
+
+  const putExpertProfile = () => {
+    if (
+      email == null ||
+      expertise == null ||
+      description == null ||
+      expertiseTags == null ||
+      visibility == null
+    ) {
+      setFieldsMissing(true);
+      return;
+    }
+    apiContext
+      .updateJufoExpert(user.id, {
+        contactEmail: email,
+        description,
+        expertiseTags: expertiseTags.split(','),
+        active: visibility,
+      })
+      .then(() => closeModal());
+  };
+
+  return (
+    <DialogModalBase accentColor={accentColor}>
+      <DialogModalBase.Modal modalName="editExpertProfileModal">
+        <DialogModalBase.Header>
+          <div className={styles.avatarWrapper}>
+            <Avatar className={styles.avatar} />
+            <button
+              className={styles.avatarOverlay}
+              onClick={() => avatarUploader.current.click()}
+            >
+              <p>Ändern</p>
+              <input
+                type="file"
+                id="file"
+                ref={avatarUploader}
+                accept="image/*"
+                style={{ display: 'none' }}
+                onChange={(e) => {
+                  e.stopPropagation();
+                  e.preventDefault();
+                  const file = e.target.files[0];
+                  console.log(file);
+                  const fileReader = new FileReader();
+                  fileReader.onload = () => {
+                    setAvatar(() => ({ className }) => {
+                      return (
+                        <img
+                          className={className}
+                          alt="avatar"
+                          src={String(fileReader.result)}
+                        />
+                      );
+                    });
+                  };
+
+                  fileReader.readAsDataURL(file);
+                }}
+              />
+            </button>
+          </div>
+        </DialogModalBase.Header>
+        <DialogModalBase.Spacer />
+        <DialogModalBase.Header>
+          <DialogModalBase.Title>Melanie Meiers</DialogModalBase.Title>
```

Comment:
This should be the user's name ;D

Reply:
Fixed in 3a687e1458e47e8e60698eadfb36d91abd70d3f1